src/core/memory.c

```diff
@@ -104,3 +123,48 @@ void *allocate_huge_page(int size)
   return NULL;
 }
 
+// SIGSEGV handler: attempts map packet memory on demand
+void allocate_on_sigsegv(int sig, siginfo_t *si, void *unused)
+{
+  uint64_t address = (uint64_t)si->si_addr;
+  if ((address & 0x500000000000ULL) != 0x500000000000ULL) {
+    // This is not DMA memory: die.
+    //exit(139);
```

Comment:
This line should probably be uncommented right? I mean: if a genuine segfault occurs then we want the process to exit with status 139.

Currently it looks like the segfault behavior has turned into an infinite loop:

```
$ sudo ./snabb snsh -e 'require("ffi").cast("int*", 0)[0] = 0'
^C
```

Sorry about the slow feedback. I think this change is in good shape. Just want to let my subconscious ponder a little bit if there are any subtle issues that we may have missed particularly regarding the memory sharing. (Can also debug these after a merge to next of course.)